### PR TITLE
Fix Issue 8608: When using tab key focus does not appear correctly on chat rows

### PR DESCRIPTION
**Pull request by andrewlor**
### Details
When using the tab key on web to cycle through focusing of elements, the outline does not appear correctly when cycling through the chat rows.
tab-focus-bug

### Fixed Issues

$ https://github.com/Expensify/App/issues/8608

### Tests

1. On web or desktop, login
2. Press tab at least 4 times to start cycling through chat rows
3. Expect outline style on a focused chat row to be visible on all 4 sides:
Screen Shot 2022-08-14 at 12 21 14 PM
- [x] Verify that no errors appear in the JS console

#### Contributor (PR Author) Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

<details>
<summary><h4>PR Reviewer Checklist</h4>

The Contributor+ will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] Any functional components have the `displayName` property
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

### QA Steps

1. On web or desktop, login
2. Press tab at least 4 times to start cycling through chat rows
3. Expect outline style on a focused chat row to be visible on all 4 sides:
Screen Shot 2022-08-14 at 12 21 14 PM
- [ ] Verify that no errors appear in the JS console

#### Web

https://user-images.githubusercontent.com/8475244/184549766-393f652c-b213-4caf-a1dc-9dbda7e9a9cb.mov

https://user-images.githubusercontent.com/8475244/184549779-5a6379fd-cbcf-4904-ad5b-ed271fe2dd1f.mov

https://user-images.githubusercontent.com/8475244/184549784-a144cd40-2ad2-42ad-97d8-d033177f49d2.mov

#### Mobile Web

https://user-images.githubusercontent.com/8475244/184549809-5f105b7e-04cb-411f-8e9a-cee7afd04a2c.mov

(expected functionality still works with tab key on emulator, not sure if it's possible to recreate functionality on real device)

#### Desktop

https://user-images.githubusercontent.com/8475244/184549819-d1eb3665-e822-4467-8744-d85b484795d3.mov

#### iOS

ios

#### Android

android

### CODE REVIEW

**Comment by Santhosh-Sellavel** [comment]
@shawnborton Just double checking, are we okay with this?

https://user-images.githubusercontent.com/85645967/184263613-e66beff8-0417-451a-8816-abf55aeed416.mov

@andrewlor Still Inputs looks like a 1px outline,

https://user-images.githubusercontent.com/85645967/184263767-428f9ba6-9e4f-4af9-bb6b-843d17bac80c.mov

cc: @Puneet-here

**Review by Santhosh-Sellavel — CHANGES_REQUESTED** [review]
Please handle this https://github.com/Expensify/App/pull/10368#issuecomment-1212616360

**Comment by andrewlor** [comment]
> Please handle this [#10368 (comment)](https://github.com/Expensify/App/pull/10368#issuecomment-1212616360)

Do we want to increase the border width of inputs to 2px? I think we'd have to make other considerations. The input focus is done through the `border` property because inputs already have a grey border when unfocused, it's just the color that changes when focused. So would we want to increase the border width of all inputs to 2px regardless of their focus state? Or would we want the border size to jump from 1px to 2px depending on whether they are focused?

The goal of this issue is to fix a bug, so I think changes to the general approach to styling is an increase in scope from the original issue.

I think the best options to fix this bug are to either keep these changes as they are, or decrease the width back to 1px to be consistent with the inputs (happy to do either). And then it might make more sense to approach the general styling of all of these things holistically in another PR.

Let me know what your thoughts are.

**Comment by Santhosh-Sellavel** [comment]
@shawnborton your thoughts

**Comment by puneetlath** [comment]
I'll defer to shawn, but I would think we should leave the inputs as they are.

**Comment by Santhosh-Sellavel** [comment]
It just looks inconsistent to me, but I prefer 1px everywhere.

**Comment by shawnborton** [comment]
I don't think we want to touch inputs or their styling. So maybe you are right, let's just use the simple 1px outline for now.

**Comment by Santhosh-Sellavel** [comment]
@andrewlor Update screen recordings

**Comment by andrewlor** [comment]
@Santhosh-Sellavel Updated screenshots / recordings

**Comment by Santhosh-Sellavel** [comment]
@Expensify/design @puneetlath 

While testing this PR I found this, tab focus border is broken. Is this an issue should? 
Also, this is not caused by this PR.

Screenshot 2022-08-15 at 11 37 07 PM

**Comment by Santhosh-Sellavel** [comment]
<h4>PR Reviewer Checklist</h4>

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by tagging the marketing team on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] Any functional components have the `displayName` property
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

**Review by Santhosh-Sellavel — APPROVED** [review]
🎀 👀 🎀 
C+ Reviewed

@puneetlath Tests well for me.

Except for this https://github.com/Expensify/App/pull/10368#issuecomment-1215542134. Most likely will be handled in a separate issue if it's a valid one.

Also, this is not caused by this PR, so approving this.

**Review by puneetlath — APPROVED** [review]

**Comment by shawnborton** [comment]
Regarding the comment above, I think that's fine for now. I think we can come back through and polish this up as another project. For example, I really do think that the tab focus should generally appear on the outside of most elements, but I see where it makes sense to do them as inset for things like chat rows. If there is an easy way to do inset for chat rows and outer shadow for all other elements, I'd be down to do that now. Otherwise we can follow up if needed.

**Comment by puneetlath** [comment]
I think we should just do it as a follow-up since that's a bit larger scope than this job.

**Review by shawnborton — APPROVED** [review]

File: web/index.html [hunk 1]
```diff
@@ -34,11 +34,11 @@
         }
         :focus-visible {
             outline: 0;
-            box-shadow: 0px 0px 0px 1px #0185ff;
+            box-shadow: inset 0px 0px 0px 1px #0185ff;
```
Comment (shawnborton) [inline web/index.html:37]: Rather than using a hex code, can we refer to a color variable instead please?

File: web/index.html [hunk 2]
```diff
@@ -34,11 +34,11 @@
...
         }
         :focus[data-focusvisible-polyfill] {
             outline: 0;
-            box-shadow: 0px 0px 0px 1px #0185ff;
+            box-shadow: inset 0px 0px 0px 1px #0185ff;
```
Comment (shawnborton) [inline web/index.html:41]: Same comment as above.

**Review by shawnborton — CHANGES_REQUESTED** [review]
Sorry, found two quick things regarding color variables that we should fix.

File: web/index.html (hunk 1 above)
Comment (andrewlor) [reply web/index.html:37]: Unfortunately I believe this suffers from the same issue we're encountering with trying to select specific components for inside/outside shadows: this is outside the scope of the main javascript source code.

So I'm not immediately sure of a way to reference a variable in `colors.js` in this pseudo selector. @Santhosh-Sellavel @puneetlath Any ideas?

**Comment by puneetlath** [comment]
@shawnborton if you approve the PR then we can merge.

**Review by shawnborton — APPROVED** [review]

**Pull request merged**